projects/align-ptr.md

```diff
@@ -0,0 +1,39 @@
+# align(ptr)
+
+## Summary and problem statement
+
+The `align` attribute can be insufficiently expressive for some types, and we aim to improve some use cases.
+
+## Prioritization
+
+This group calls under the "Targeted ergonomic wins and extensions" priority.
+
+## Motivation, use-cases, and solution sketches
+
+It's sometimes the case that a type must be aligned based on the platform's pointer width. For example, the `AtomicPtr` type is defined as follows:
+
+```rust
+#[cfg_attr(target_pointer_width = "16", repr(C, align(2)))]
+#[cfg_attr(target_pointer_width = "32", repr(C, align(4)))]
+#[cfg_attr(target_pointer_width = "64", repr(C, align(8)))]
+pub struct AtomicPtr<T> {
+    p: UnsafeCell<*mut T>,
+}
```

Comment:
Why is `AtomicPtr` defined like this? 

`UnsafeCell` is `#[repr(transparent)]` and `AtomicPtr` is `#[repr(C)]`; my understanding is that the minimum alignment of `AtomicPtr` therefore must be equal to the minimum alignment of `*mut _` . 

Unless there's a platform in which the minimum alignment of `AtomicPtr` must be greater than the minimum alignment of `*mut _`, the  `cfg_attr`s and `align(*)` modifiers seem redundant: `AtomicPtr` could be annotated with just `#[repr(C)]`.

Am I missing something?

Reply:
I don't think you're missing anything.